File: model/task.go
package model

// Task represents a single task in a project.
type Task struct {
	ID        int64
	ProjectID int64
	TaskUpdate
}

// TaskUpdate represents an update of the mutable fields of a Task.
type TaskUpdate struct {
	EpicID      int64
	AssigneeID  int64
	StateID     int64
	Title       string
	Description string
}

// NewTask returns an instantiated Task.
func NewTask(projectID, epicID, assigneeId, stateID int64, title, description string) *Task {
	return &Task{
		ProjectID: projectID,
		TaskUpdate: TaskUpdate{
			EpicID:      epicID,
			AssigneeID:  assigneeId,
			StateID:     stateID,
			Title:       title,
			Description: description,
		},
	}
}

// Update updates this Task's mutable fields with the given update.
func (m *Task) Update(update TaskUpdate) {
	m.EpicID = update.EpicID
	m.AssigneeID = update.AssigneeID
	m.StateID = update.StateID
	m.Title = update.Title
	m.Description = update.Description
}

File: frontend/src/components/Task.js
/**
 * Task represents a single unit of work on a board.
 */
export default class Task {
  /**
   * Task constructor.
   * @param {Number} id ID Unique ID of object.
   * @param {Number} projectId ID of the project this task is part of.
   * @param {Number} epicId ID of the epic this task is in, or 0 otherwise.
   * @param {Number} assigneeId ID of assignee of this task.
   * @param {Number} stateId ID of state of this task.
   * @param {String} title Title of task.
   * @param {String} description Description of task.
   */
  constructor(id, projectId, epicId, assigneeId, stateId, title, description) {
    this.id = id;
    this.projectID = projectId;
    this.epicID = epicId;
    this.assigneeID = assigneeId;
    this.stateId = stateId;
    this.title = title;
    this.description = description;
  }
}
